docs/src/tutorials/Nonlinear programs/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,331 @@
+#                                                                                 #src
+# The above copyright notice and this permission notice shall be included in all  #src
+# copies or substantial portions of the Software.                                 #src
+#                                                                                 #src
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      #src
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   #src
+# SOFTWARE.                                                                       #src
+
+# # Space Shuttle Reentry Trajectory
+
+# **Originally Contributed by**: Henrique Ferrolho
+
+# This tutorial demonstrates how to compute a reentry trajectory for the
+# [Space Shuttle](https://en.wikipedia.org/wiki/Space_Shuttle),
+# by formulating and solving a nonlinear programming problem.
+# The problem was drawn from Chapter 6 of 
+# ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
+# by John T. Betts.
+
+# !!! tip
+#     This tutorial is a more-complicated version of the [Rocket Control](@ref) example.
+#     If you are new to solving nonlinear programs in JuMP, you may want to start there instead.
+
+# The motion of the vehicle is defined by the following set of DAEs:
+# 
+# ```math
+# \begin{aligned}
+# \dot{h} & = v \sin \gamma , \\
+# \dot{\phi} & = \frac{v}{r} \cos \gamma \sin \psi / \cos \theta , \\
+# \dot{\theta} & = \frac{v}{r} \cos \gamma \cos \psi , \\
+# \dot{v} & = -\frac{D}{m} - g \sin \gamma , \\
+# \dot{\gamma} & = \frac{L}{m v} \cos(\beta) + \cos \gamma \left ( \frac{v}{r} - \frac{g}{v} \right ) , \\
+# \dot{\psi} & = \frac{1}{m v \cos \gamma} L \sin(\beta) + \frac{v}{r \cos \theta} \cos \gamma \sin \psi \sin \theta , \\
+# q & \le q_U , \\
+# \end{aligned}
+# ```
+
+# where the aerodynamic heating on the vehicle wing leading edge is $q = q_a q_r$
+# and the dynamic variables are
+# 
+# ```math
+# \begin{aligned}
+#      h & \quad \text{altitude (ft)},     \qquad & & \gamma \quad \text{flight path angle (rad)}, \\
+#   \phi & \quad \text{longitude (rad)},   \qquad & & \psi   \quad \text{azimuth (rad)},           \\
+# \theta & \quad \text{latitude (rad)},    \qquad & & \alpha \quad \text{angle of attack (rad)},   \\
+#      v & \quad \text{velocity (ft/sec)}, \qquad & & \beta  \quad \text{bank angle (rad)}.
+# \end{aligned}
+# ```
+
+# The aerodynamic and atmospheric forces on the vehicle are
+# specified by the following quantities (English units):
```

Comment:
Is "English unit" a standard term? I would call `ft` and `lb` etc "Imperial," but that may just be a Kiwi thing...

Reply:
In Scotland we also call them imperial units. It just happens to be how the problem is originally stated in the book; see previous discussion about this https://github.com/jump-dev/JuMPTutorials.jl/pull/97#discussion_r572706571.